Source/Core/InputCommon/ControllerInterface/OSX/OSXJoystick.h

```diff
@@ -83,6 +85,9 @@ class Joystick : public ForceFeedback::ForceFeedbackDevice
 private:
   const IOHIDDeviceRef m_device;
   const std::string m_device_name;
+  std::vector<IOHIDElementCookie> m_cookies;
```

Comment:
This doesn't make sense as a member variable as it's only needed during construction. You could declare it in the ctor and pass it to `AddElements` as `known_cookies`. Also your usage would maybe be more suited by a `std::set` as you only check for duplicates. The long `std::find_if(...)` call would then just be `known_cookies.count(cookie)`.